pkg/controller/routerservice/router_service_controller.go

```diff
@@ -48,10 +49,28 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	// Only filter on services in the openshift-ingress namespace and create/update events
 	p := predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
-			return e.Meta.GetNamespace() == RouterServiceNamespace
+			object, err := meta.Accessor(e.Object)
+			if err != nil {
+				return false
```

Comment:
I guess you added this condition because you saw errors happening here? Are those errors worth logging?

Reply:
no. looks like `Meta` is no longer an available option for `event.CreateEvent`
`e.Meta undefined (type event.CreateEvent has no field or method Meta)`
`Meta` seems migrated to its own package, the accessor takes the e.Object as param to be able to get its Namespace.
```go
package meta ("k8s.io/apimachinery/pkg/api/meta")
```

However, if this can be refactored to be simplified, happy to refactor here.

Reply:
Gotcha.

I think this simplification works:

```go
			return e.Object.GetNamespace() == RouterServiceNamespace
```

Reply:
@2uasimojo yeah, agreed, this has been simplified now